Update OIDC documentation adding a warning for Okta configuration (#1244)
giantswarm/docs: src/content/advanced/configure-dex-in-your-cluster/index.md

data:
  values: |
    isWorkloadCluster: true
    services:
      kubernetes:
        api:
          caPem: |
            -----BEGIN CERTIFICATE-----
            M...=

>             -----END CERTIFICATE-----

Author: this is now automatically injected in last versions

Reviewer: ❤️

        connectors:
        - id: customer
          connectorName: test
          connectorType: github
          connectorConfig: >-
            clientID: <CLIENT-ID-SET-IN-YOUR-IdP>
            clientSecret: <CLIENT-SECRET-SET-IN--YOUR-IdP>
            loadAllGroups: false
            redirectURI: https://dex.<CLUSTERID>.<BASEDOMAIN>/callback
            orgs:
            - name: main-customer-org
              teams:

>               - team-infra

Reviewer: Why this change?

Reviewer: Oh, I see. It's supposed to be the GitHub example, but used the wrong `connectorType`.

The rest with `loadAllGroups: false` however I'd still like to understand. Why not load all group memberships?

Author: yeah it was a mistake and I have just realized. 

IMO it is a good default, so define which orgs you want to give access instead of pulling all

Reviewer: OK for me

            redirectURI: https://dex.<CLUSTERID>.<BASEDOMAIN>/callback
```

{{< /tab >}}
{{< tab title="Okta">}}

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: dex-app-user-values
  namespace: <CLUSTERID>
data:
  values: |
    isWorkloadCluster: true
    oidc:
      expiry:
        signingKeys: 6h
        idTokens: 30m
      customer:
        enabled: true
        connectors:
        - id: customer
          connectorName: test
          connectorType: oidc
          connectorConfig: >-
            clientID: <CLIENT-ID-SET-IN-YOUR-IdP>
            clientSecret: <CLIENT-SECRET-SET-IN--YOUR-IdP>

>             insecureEnableGroups: true

Author: ```suggestion
            insecureEnableGroups: true
            insecureSkipEmailVerified: true
```

            getUserInfo: true
            scopes:
            - email
            - groups
            - profile
            issuer: https://<OKTA_OIDC_ENDPOINT>

>             redirectURI: https://dex.<CLUSTERID>.<BASEDOMAIN>/callback

Reviewer 2: `insecureSkipEmailVerified: true` is missng

```

{{< /tab >}}
{{< /tabs >}}

> __Warning__: With `oidc` connector you might need to add `getUserInfo` in the connector configuration to force a second called to IdP in order to get groups (for example required by Okta). More info [here](https://github.com/dexidp/dex/issues/1065).

Reviewer: ```suggestion
__Warning__: With `oidc` connector you might need to add `getUserInfo` in the connector configuration to force a second call to the identity provider in order to get groups. This is required for example by Okta. More info on this can be found in [dexipd/dex#1065](https://github.com/dexidp/dex/issues/1065).
```

            redirectURI: https://dex.<CLUSTERID>.<BASEDOMAIN>/callback
```

{{< /tab >}}
{{< tab title="Okta">}}

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: dex-app-user-values
  namespace: <CLUSTERID>
data:
  values: |
    isWorkloadCluster: true
    oidc:
      expiry:
        signingKeys: 6h
        idTokens: 30m
      customer:
        enabled: true
        connectors:
        - id: customer
          connectorName: test
          connectorType: oidc
          connectorConfig: >-
            clientID: <CLIENT-ID-SET-IN-YOUR-IdP>
            clientSecret: <CLIENT-SECRET-SET-IN--YOUR-IdP>
            insecureEnableGroups: true

>             insecureSkipEmailVerified: true

Reviewer 2: Sorry @pipo02mix I didn't realise this is a generic guide. In that case `insecureSkipEmailVerified` should go to the warning section. I think same applies to `insecureEnableGroups`. I'm not even sure if we needed to add this in our last case.

>             insecureSkipEmailVerified: true

Author: ```suggestion
```
